packages/cdk8s/lib/app.ts

```diff
@@ -36,8 +40,32 @@ export class App extends Construct {
       fs.mkdirSync(this.outdir);
     }
 
-    Node.of(this).synthesize({
-      outdir: this.outdir
-    });
+    // create explicit chart dependencies
+    // from implicit construct dependencies
+    for (const dep of Node.of(this).dependencies) {
+
+      const sourceChart = Chart.of(dep.source);
+      const targetChart = Chart.of(dep.target);
+
+      if (sourceChart !== targetChart) {
+        Node.of(sourceChart).addDependency(targetChart);
+      }
+    }
+
+    // TODO add validate call here
+
+    const charts: IConstruct[] = new DependencyGraph(Node.of(this)).topology().filter(x => x instanceof Chart);
+
+    for (const index in charts) {
+
+      const node: IConstruct = charts[index];
+      const chart: Chart = Chart.of(node);
+      const manifestFile = `${Node.of(chart).uniqueId}.k8s.yaml`;
+
+      const paddedIndex = index.toString().padStart(4, '0');
+
+      Yaml.save(path.join(this.outdir, `${paddedIndex}-${manifestFile}`), chart.toJson());
```

Comment:
I think we shouldn't add the prefix if there is only a single chart or if there are no dependency relationship between charts.

Reply:
👍